scripts/ci/publish.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# Based on: https://github.com/tokio-rs/tokio/blob/master/bin/publish
+
+set -e
+USAGE="Publish a new release of a lighthouse crate
+USAGE:
+    $(basename "$0") [OPTIONS] [CRATE_PATH] [CRATE] [VERSION]
+OPTIONS:
+    -v, --verbose       Use verbose Cargo output
+    -d, --dry-run       Perform a dry run (do not publish the release)
+    -h, --help          Show this help text and exit
+        --allow-dirty   Allow dirty working directories to be packaged"
+
+DRY_RUN=""
+DIRTY=""
+VERBOSE=""
+
+verify() {
+    echo "Verifying if $CRATE v$VERSION can be released"
+    ACTUAL=$(cargo pkgid | sed -n 's/.*:\(.*\)/\1/p')
```

Comment:
I think the reason for this is that `cargo pkgid` has a different format if the package has a different `[lib]` name and `[package]` name. Becasue the `sed` expression tries to find the version based on the `:`. I didn't run into this with my crates because of the `sean_test_` prefix I only added to the `[package]` name.

`file:///home/sean/IdeaProjects/lighthouse/consensus/tree_hash_derive#0.3.0`
`file:///home/sean/IdeaProjects/lighthouse/consensus/serde_utils#eth2_serde_utils:0.1.0`

Should we try to make `[lib]` name = `[package]` name? Not sure what the benefit of having them different is. Or I can update the expression to look for the `:` first and fall back to the `#`.